R/utils.R

```diff
@@ -144,3 +144,18 @@ convert_to_logsd <- function(mean, sd) {
 to_simplex <- function(vector) {
   return(vector / sum(vector))
 }
+
+#' Escape brackets returned in a string for passing to glue
+#'
+#' @param string A string vector containing `{}`
+#'
+#' @return A string vector where all single brackets are replaced with double
+#' brackets
+#' @export
+#'
+#' @examples
+#' test_string <- "{my_var}"
+#' glue::glue(autoescape_brackets(test_string))
+autoescape_brackets <- function(string) {
+  return(gsub("\\{(.*)\\}", "{{\\1}}", string))
+}
```

Comment:
Shouldn't this be an internal function?

Reply:
Agree, I would not export this.

Reply:
I might have needed it for an example of something I was exporting (which probably also could be an external function-- let me look!)

Reply:
I'm guessing it's the example you wrote here for the function itself. The catch is that internal functions typically [don't have examples](https://github.com/tidyverse/dplyr/blob/main/R/utils.R).

```R
> test_string <- "{my_var}"
> glue::glue(autoescape_brackets(test_string))
Error in autoescape_brackets(test_string) : 
  could not find function "autoescape_brackets"
```